packages/playground/remote/service-worker.ts

```diff
@@ -25,6 +29,57 @@ if (!(self as any).document) {
 	self.document = {};
 }
 
+/**
+ * Forces the browser to always use the latest service worker.
+ *
+ * Each service worker build contains a hardcoded `buildVersion` used to derive a cache key
+ * for offline-mode-cache. As long as the previous service worker is used, it will
+ * keep serving a stale version of Playground assets, e.g. `/index.html`, `php.wasm`, etc.
+ *
+ * This is problematic for two reasons:
+ *
+ * 1. Users won't receive critical bugfixes for up to 24 hours after they're released [1].
+ * 2. Users will experience fatal crashes. Assets such as the WebAssembly PHP builds are
+ *    loaded asynchronously using fetch() and import() functions. The specific URLs are
+ *    hardcoded by the bundler at build time, e.g. the worker-thread.js file contains
+ *    a call similar to `import("./assets/php_8_3-2286e20c.js")`. If the browser uses
+ *    a stale version of the worker thread, it will try to import a JavaScript file
+ *    that no longer exists.
+ *
+ * See also: https://github.com/WordPress/wordpress-playground/issues/105
+ *
+ * [1] https://web.dev/articles/service-worker-lifecycle#updates
+ */
+self.addEventListener('install', (event) => {
+	/**
+	 * Skip over the "waiting" lifecycle state, to ensure that our
+	 * new service worker is activated as soon as possible while respecting user data.
+	 */
+	event.waitUntil(
+		self.clients
+			.matchAll({
+				type: 'window',
+			})
+			.then((windowClients) => {
+				const clientsThatCanBeSafelyRefreshed = windowClients.filter(
+					() => {
+						// @TODO: Only if we're running a version of Playground that
+						//        stores temporary sites in OPFS. Otherwise we'd
+						//        destroy the user's in-memory changes.
```

Comment:
Even if all sites are being saved to OPFS, users could still have unsaved, in-memory changes while editing posts or something similar. Because of this, we might just need to ask users via prompts in Playground instances (#566).

Reply:
It's not much of a choice, though. If the user says "no" in the prompt, their running app will be broken as they will no longer be able to, say, start a new site with another version of PHP as the relevant .wasm will no longer be available on the server. I'd rather make this "just work" and find a way to force refresh without losing data. One way of doing that would be storing all the "temporary" sites in OPFS anyway, only showing the ones created in the current browsing session, and cleaning the rest after 24h.

Reply:
The more I think about this, the more I feel that the installation and update process needs to be fully controlled. 
If you would always get the cached version first we would ensure nothing breaks on the first load. 
After that, we can check for an update async and ask the user what to do.

I would do it automatically if it wasn't for their data, but we can't just remove someone's work just because it's not saved. (Imagine using Playground as your primary note taking app and it randomly reloads)

Reply:
What kind of data loss are we risking? Posts have autosaves and revisions.